usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateBothRouteNamesPresence.java

```diff
@@ -0,0 +1,82 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.routes.Route;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.MissingBothRouteNamesNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.warning.MissingRouteLongNameNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.warning.MissingRouteShortNameNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.Collection;
+
+/**
+ * Use case to validate that Route short name and long name are present.
+ */
+public class ValidateBothRouteNamesPresence {
+
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     */
+    public ValidateBothRouteNamesPresence(final GtfsDataRepository dataRepo,
+                                          final ValidationResultRepository resultRepo) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+    }
+
+    /**
+     * Use case execution method: checks if Route short name and long name are missing
+     * for every Routes in a {@link GtfsDataRepository}. If both are missing, a new error notice is generated.
+     * If only one is missing, a warning error notice is generated. This notice is then added
+     * to the {@link ValidationResultRepository} provided in the constructor.
+     */
+    public void execute() {
+        Collection<Route> routes = dataRepo.getRouteAll();
+        routes.stream()
+                .filter(route -> !(isPresentLongName(route.getRouteLongName()) && isPresentShortName(route.getRouteShortName())))
+                .forEach(route -> {
+                    if (!isPresentLongName(route.getRouteLongName()) && !isPresentShortName(route.getRouteShortName())) {
+                        resultRepo.addNotice(new MissingBothRouteNamesNotice("route.txt", route.getRouteId()));
+                    } else if (!isPresentLongName(route.getRouteLongName())) {
+                        resultRepo.addNotice(new MissingRouteLongNameNotice("route.txt", route.getRouteId()));
+                    } else {
+                        resultRepo.addNotice(new MissingRouteShortNameNotice("route.txt", route.getRouteId()));
+                    }
+                });
+    }
+
+    /**
+     * @param routeShortName the short name of a Route
+     * @return true if Route short name is not missing or empty, false if not.
+     */
+    private boolean isPresentShortName(final String routeShortName) {
+        return routeShortName != null && !routeShortName.replaceAll("\\s+", "").isEmpty();
```

Comment:
Can use guava `Strings.isNullOrEmpty`

Reply:
Is the goal here to check if the `routeShortName` is only whitespace, and if so consider that to be an empty string?

Then, I think you probably want something like this if you're using pure Java:

```java
return routeShortName != null && !routeShortName.isBlank();
```

`isBlank()` is new as of Java 11, and IIRC we're requiring Java 11 so we should be ok using this.

Alternately, here's the version using Guava (building on @fabrice-v comment):

```java
return !Strings.isNullOrEmpty(routeShortName) && !routeShortName.isBlank();
```

Both of the above should be faster than compiling a regex on each invocation and IMHO is easier to read too (assuming you're not familiar with what `"\\s+"` means). I'd leave towards the first just because `isBlank()` already checks for `isEmpty()`, so you're duplicating that step in the 2nd example.

Again, it would be nice to have unit tests for these types of methods that have examples of expected input and output - otherwise later we may have to guess as to what this method was supposed to do.

Reply:
I kept 
```return routeShortName != null && !routeShortName.isBlank();```
I also added unit tests